### DISCUSSION TITLE
Settings Profiles with option to auto-apply them per ruleset

### DISCUSSION THREAD

**Discussion by LeaPhant** [ppy/osu#22988] [Ideas]
Currently most settings are applied across all rulesets apart from exceptions like keybinds or skinning which supports customization per ruleset. As someone who likes playing different modes and also switch between them in multiplayer it's important to me to be able to have my comfortable settings for each mode (which also includes using the skins I'm used to for each mode without having to manually put them together into one skin somehow).

I've come up with an idea for a solution to this which could also offer easier management of one osu install for different situations: **Profiles**.

![image](https://user-images.githubusercontent.com/14080165/227800937-6154b9c1-f8a9-42e3-a56c-b97381f27bdd.png)

This would be a section in the settings menu where you can:
- Create named profiles which each store the setting you've changed compared to the default
- Move them up or down with the uppermost profile overwriting the ones below
- Select a ruleset for a profile which gets automatically applied if the ruleset is selected (and unapplied if another is selected)

Possible use cases:
- Quickly turning off more graphically intense effects and applying a lower fps limit in case you want to preverse battery
- Automatically limiting to Vsync for taiko but keeping 8x refresh rate for other modes
- Using different tablet areas or mouse sensitivities depending on ruleset (this would be more relevant once there's more than one ruleset that uses these inputs apart from e.g. relax osu!catch)

I have found [previous](https://github.com/ppy/osu/discussions/21898#discussioncomment-4631473) [comments](https://github.com/ppy/osu/discussions/21362#discussioncomment-4196178) on profiles/presets being a planned feature but I still decided to go forward with posting this since I want to put an emphasis on allowing these presets to be automatically applied based on predicates, starting with ruleset.

**Comment by peppy** [comment]
None of the use cases make sense to me to be per ruleset except maybe the mouse sensitivity.

Even in such a use case, it's so rare I would not be interested in making this automatic. Too much complexity. if you want that level of profiles, you can switch yourself (we can make a hotkey to switch or something).

**Reply by officialrestaurant** [reply]
Writing my request here to not create a potential duplicate, even though my request is not per gamemode but per osu!std mod.

It's very rare to find an osu! player that doesn't switch skins for DT. If it was possible to set a DT skin, an EZ skin etc per-mod, it would be used by most people who play for performance. The biggest effect would be in multiplayer/tourneys, as LeaPhant mentioned above.
